modelzoo/test/test_modelzoo.py

```diff
@@ -0,0 +1,109 @@
+import math
+import operator
+import os
+from functools import reduce
+from uuid import UUID
+
+import numpy as np
+import pytest
+from PIL import Image
+
+from modelzoo.admin import *
+from modelzoo.exceptions import *
+from modelzoo.protos.services_pb2 import (Image, Payload, PayloadType, Table,
+                                          Text)
+from modelzoo.utils import *
+
+
+# Utils for testing
+def compare_image(im1, im2):
+    h1 = im1.histogram()
+    h2 = im2.histogram()
+
+    rms = math.sqrt(
+        reduce(operator.add, map(lambda a, b: (a - b) ** 2, h1, h2)) / len(h1)
+    )
+    return rms
+
+
+def validate_uuid(uuid):
+    try:
+        uuid_obj = UUID(uuid_to_test, version=4)
+    except ValueError:
+        return False
+
+    return str(uuid_obj) == uuid_to_test
+
+
+# Test ModelZoo Utils
+def test_image_utils():
+    im_arr = np.random.rand(100, 100, 3) * 255
+    im = Image.fromarray(im_arr.astype("uint8")).convert("RGB")
+    im.save("a.jpg")
+    uri1 = image_input_types_to_uri(im)
+    uri2 = image_input_types_to_uri("a.jpg")
+    uri3 = image_input_types_to_uri(uri1)
+    assert (
+        validate_data_uri(uri1) and validate_data_uri(uri2) and validate_data_uri(uri3)
+    )
+    assert uri1 == uri3
+    im1 = uri_to_img(uri1)
+    im2 = uri_to_img(uri2)
+    assert (
+        compare_image(im1, im2) < 5
+    )  # Magic Numbers. Found by testing the result, and finding margins.
+    # Difference since saving with PIL causes slight image quality reduction.
+    assert compare_image(im1, im) < 24  # Same as above.
+    uri_to_img_file(uri3, "b.jpg")
+    im3 = Image.open("b.jpg")
+    assert (
+        compare_image(im3, im2) == 0
+    )  # Should be the same image. Saving image via PIL causes slight quality reduction.
+    os.remove("a.jpg")
+    os.remove("b.jpg")
+
+
+# Test ModelZoo Connection
+def test_modelzoo_connection():
```

Comment:
this seems to require go server and python server to be up? Like if I just run `test_modelzoo.py` it will fail right?

Reply:
```python
@pytest.fixture
def start_modelzoo_service():
    go_proc = subprocess.Popen("GO binary serve --port y")
    clipper_mock_proc = subprocess.Popen("gunicorn serve clipper:app at port x")
    yield
    go_proc.kill()
    clipper_mock_proc.kill()
```